keda/templates/12-keda-deployment.yaml

```diff
@@ -39,6 +39,7 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ .Values.serviceAccount.name }}
+      automountServiceAccountToken: true
```

Comment:
Can you move this to a value with default property `true` instead please?

Reply:
Yes I can do that. I am not sure if this makes sense as you need that to set to true for having API Access. If you would set it to false KEDA would not work anymore. Please confirm if I should change it.

Reply:
With that you mean if `serviceAccount.automountServiceAccountToken` flag is on true? If so, why not re-use that same value then?

Reply:
Let me explain:
Every ServiceAccount generates an API Token regardless of what automountServiceAccountToken is set to (true, false or even not set)

Default behaviour when not specified automountServiceAccountToken in ServiceAccount and Deployment -> Pods using this ServiceAccount can access the API-Token.

ServiceAccount.automountServiceAccountToken: false -> All pods using this ServiceAccount have no access to API-Token anymore (but the ServiceAccount still have an API-Token stored internally)

Deployment can opt-in to regain access to API-Token:
Deployment.automountServiceAccountToken: true -> pods regain access to token

Does a switch make sense for ServiceAccount.automountServiceAccountToken?
Yes because e.g. we have a security policy that we can only deploy ServiceAccounts with ServiceAccount.automountServiceAccountToken: false.
So we have to define Deployment.automountServiceAccountToken: true in every Deployment which needs API-Access.

If somebody would use the ServiceAccount for own custom Deployments without specifying Deployment.automountServiceAccountToken they need to set ServiceAccount.automountServiceAccountToken: true. (Which would be the default switch value... So they do not recognize a change)

Does a switch make sense for Deployment.automountServiceAccountToken?
No. Because KEDA needs API-Access. A switch which could possible set Deployment.automountServiceAccountToken: false will break the API-Access for KEDA-Pods. (Regardless of ServiceAccount.automountServiceAccountToken)

Why even specifying Deployment.automountServiceAccountToken?
Because when ServiceAccount.automountServiceAccountToken: false and not specifying Deployment.automountServiceAccountToken the API-Access is gone.

Why not re-use the same value?
Because when setting ServiceAccount.automountServiceAccountToken: false then reusing value would result in Deployment.automountServiceAccountToken: false which prevents API-Access for pods.

Reply:
Thanks for explaining!